sdk/python/feast/data_source.py

```diff
@@ -441,26 +443,32 @@ class RequestSource(DataSource):
 
     Args:
         name: Name of the request data source
-        schema: Schema mapping from the input feature name to a ValueType
+        schema Union[Dict[str, ValueType], List[Field]]: Schema mapping from the input feature name to a ValueType
         description (optional): A human-readable description.
         tags (optional): A dictionary of key-value pairs to store arbitrary metadata.
         owner (optional): The owner of the request data source, typically the email of the primary
             maintainer.
     """
 
     name: str
-    schema: Dict[str, ValueType]
+    schema: Union[Dict[str, ValueType], List[Field]]
```

Comment:
Maybe the internals should always be using `List[Field]`, and we convert to that in `__init__` and `from_proto`? Or is this field exposed to users directly, and it would break them if we changed it's type?

Reply:
Actually looking at the `from_proto` method we're changing the type of this field anyway

Reply:
Yup, removing union.